include/pmacc/cuSTL/algorithm/mpi/Reduce.tpp

```diff
@@ -80,11 +80,13 @@ Reduce<dim>::Reduce(const zone::SphericZone<dim>& p_zone, bool setThisAsRoot) :
 
     // avoid deadlock between not finished pmacc tasks and mpi blocking collectives
     __getTransactionEvent().waitForFinished();
-    MPI_CHECK(MPI_Comm_group(MPI_COMM_WORLD, &world_group));
-    MPI_CHECK(MPI_Group_incl(world_group, new_ranks.size(), &(new_ranks.front()), &new_group));
-    MPI_CHECK(MPI_Comm_create(MPI_COMM_WORLD, new_group, &this->comm));
-    MPI_CHECK(MPI_Group_free(&new_group));
-    MPI_CHECK(MPI_Group_free(&world_group));
+    if(new_ranks.size())
+    {
+        MPI_CHECK(MPI_Group_incl(world_group, new_ranks.size(), &(new_ranks.front()), &new_group));
```

Comment:
Why did you delete `MPI_CHECK(MPI_Comm_group(MPI_COMM_WORLD, &world_group));` in the line before?

Reply:
Ah, good catch. My mistake, it should stay in place.

Reply:
Fixed and rebased. Also moved declarations of those two variables inside the `if` statement as they are only used there.

Reply:
That would have been my next question ^^